### PR TITLE
#192 Ensure VanillaBytes don't require 3x capacity. 

### LINKED ISSUE #193: Add a test
Add a test to Chronicle-Bytes for this use case e.g.

        final String SYMBOL_STR = "LCOM1";
        final Bytes SYMBOL = Bytes.allocateDirect(SYMBOL_STR.length());
        SYMBOL.clear();
        SYMBOL.append(SYMBOL_STR);

### PR DESCRIPTION
**Pull request by jkolobok**

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/bytes/VanillaBytes.java [hunk 1]
```diff
@@ -600,7 +600,7 @@ public int byteCheckSum(int start, int end)
         @Override
         public Bytes<Underlying> appendUtf8(char[] chars, int offset, int length)
                 throws BufferOverflowException, IllegalStateException, BufferUnderflowException, IllegalArgumentException {
-            ensureCapacity(writePosition() + length * 3L);
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/bytes/VanillaBytes.java:603]: Nice fix. `length` is the length in chars not bytes, although `offset` is in bytes I believe. Maybe we need to specify this in javadoc?

**Review by JerryShea — APPROVED** [review]

File: src/main/java/net/openhft/chronicle/bytes/VanillaBytes.java (hunk 1 above)
Comment (jkolobok) [reply src/main/java/net/openhft/chronicle/bytes/VanillaBytes.java:603]: yeah. you are right. it doesn't look clear... Thinking...

**Comment by peter-lawrey** [comment]
If there isn't plenty of space i.e. 3 * length, we can find the length it will encode it as with `AppendableUtil.findUtf8Length(chars)` and this can be used to ensureCapacity instead.

File: src/main/java/net/openhft/chronicle/bytes/VanillaBytes.java (hunk 1 above)
Comment (JerryShea) [reply src/main/java/net/openhft/chronicle/bytes/VanillaBytes.java:603]: May need to make the same fix to net.openhft.chronicle.bytes.UncheckedNativeBytes#appendUtf8

**Review by JerryShea — APPROVED** [review]

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.21ea48](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.21ea48), [BOM-2.21ea131](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.21ea131)

**Pull request merged**